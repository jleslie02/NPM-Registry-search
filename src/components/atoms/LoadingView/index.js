/** @jsx jsx */
// eslint-disable-next-line no-unused-vars
import React from 'react';
import { jsx, css, keyframes } from '@emotion/core';
import PropTypes from 'prop-types';

const propTypes = {
  loading: PropTypes.bool
};

const loading = keyframes`
  from {
    left: 50%; width: 0;z-index:100;
  }
  33.3333% {
    left: 0; width: 100%;z-index: 10;
  }
  to {
    left: 0; width: 100%;
  }
`;

export const StatelessLoadingView = props => {
  const loadingStyle = css(
    (() => ({
      position: 'relative',
      width: '100%',
      height: '6px',
      backgroundImage: `linear-gradient(to right ,rgba(82,82,82,0.87),rgba(164, 230, 77, 0.67))`,
      '.bar': {
        content: `""`,
        display: 'inline',
        position: 'absolute',
        width: '0',
        height: '100%',
        left: '50%',
        textAlign: 'center',
        ':nth-child(1)': {
          backgroundImage: `linear-gradient(to right ,rgba(115, 109, 109, 0.87),rgba(255, 134, 10, 0.97))`,
          animation: `${
            props.loading
              ? `${loading} 2s linear 1s infinite`
              : 'none'
          }`
        },
        ':nth-child(2)': {
          backgroundImage: `linear-gradient(to right ,rgba(107, 97, 97, 0.87),rgba(255, 226, 21, 0.97))`,
          animation: `${
            props.loading
              ? `${loading} 2s linear 2s infinite`
              : 'none'
          }`
        }
      }
    }))()
  );

  return (
    <div className="load-bar" css={loadingStyle}>
      <div className="bar" />
      <div className="bar" />
      <div className="bar" />
    </div>
  );
};

StatelessLoadingView.propTypes = propTypes;
StatelessLoadingView.defaultProps = {
  loading: false
};

export default StatelessLoadingView;
