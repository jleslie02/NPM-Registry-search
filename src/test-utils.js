import React from "react";
import { render } from "react-testing-library";
import makeAtomicTheme from "./theme";

export const themedRender = (MyComponent, props) => {
  return render(<MyComponent theme={makeAtomicTheme()} {...props} />);
};

export const themedComponent = (MyComponent, props) => {
  return <MyComponent theme={makeAtomicTheme()} {...props} />;
};
