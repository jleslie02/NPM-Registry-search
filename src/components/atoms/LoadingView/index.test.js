// Static component with conditional rendering so snapshot test should suffice
import renderer from "react-test-renderer";
import LoadingView from "./index";
import { themedComponent } from "../../../test-utils";

it("renders correctly with full props", () => {
  const props = {
    loading: true,
  };
  const tree = renderer.create(themedComponent(LoadingView, props)).toJSON();
  expect(tree).toMatchSnapshot();
});
