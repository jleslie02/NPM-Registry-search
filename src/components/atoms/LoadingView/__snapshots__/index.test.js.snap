// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with full props 1`] = `
<div
  className="load-bar css-lbi6ix-StatelessLoadingView"
  data-testid="loadingView"
>
  <div
    className="bar"
  />
  <div
    className="bar"
  />
  <div
    className="bar"
  />
</div>
`;
